integrated_channels/sap_success_factors/transmitters/courses.py

```diff
@@ -18,14 +20,35 @@ class SuccessFactorsCourseTransmitter(SuccessFactorsTransmitterBase):
     This endpoint is intended to carry out an export of course data to SuccessFactors for a given Enterprise.
     """
 
-    def transmit(self, payload):
+    def transmit(self, course_exporter):
         """
         Send a course data import call to SAP SuccessFactors using the client.
 
         Args:
-            payload (SapCourseExporter): The OCN course exporter object to send to SAP SuccessFactors
+            course_exporter (SapCourseExporter): The OCN course exporter object to send to SAP SuccessFactors
         """
-        serialized_payload = payload.get_serialized_data()
+
+        CatalogTransmissionAudit = apps.get_model(  # pylint: disable=invalid-name
+            app_label='sap_success_factors',
+            model_name='CatalogTransmissionAudit'
+        )
+
+        try:
+            last_catalog_transmission = CatalogTransmissionAudit.objects.filter(
+                error_message='',
+                enterprise_customer_uuid=self.enterprise_configuration.enterprise_customer.uuid
+            ).latest('created')
+        except ObjectDoesNotExist:
+            last_catalog_transmission = None
+
+        if last_catalog_transmission:
+            last_audit_summary = json.loads(last_catalog_transmission.audit_summary)
+        else:
```

Comment:
This could be switched to a single try/except/else block, like so:

```python
try:
    last_catalog_transmission = CatalogTransmissionAudit.objects.filter(
        error_message='',
        enterprise_customer_uuid=self.enterprise_configuration.enterprise_customer.uuid,
    ).latest('created')
except ObjectDoesNotExist:
    last_audit_summary = {}
else:
    last_audit_summary = json.loads(last_catalog_transmission.audit_summary)
```

The long and the short of it is that if there's no caught exception, the else block is executed immediately after the try block. It's a little odd, but not quite as odd as the `for-else` construction.

As a side note, this logic will likely need to change, post-#79. The pagination and possibility of errors on one page but not others makes things all the more complicated.